Godeps/_workspace/src/github.com/ipfs/go-datastore/multi/multi.go

```diff
@@ -0,0 +1,204 @@
+// Package mount provides a Datastore that has other Datastores
+// mounted at various key prefixes and is threadsafe
+package multi
+
+import (
+	"errors"
+	"io"
+
+	ds "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/ipfs/go-datastore/query"
+)
+
+var (
+	ErrNoMount = errors.New("no datastore mounted for this key")
+)
+
+// Note: The advance datastore is at index 0 so that it is searched first in Get and Has
+
+func New(adv ds.Datastore, normal ds.Datastore, aux []ds.Datastore, roAux []ds.Datastore) *Datastore {
```

Comment:
We should make a 'type switched datastore' with this. That way we can say things like "store all `[]byte` in datastore X" and "store all `*filestore.FileStore` in datastore Y"

I did a sketch to check if this was possible (it is, yay!)
```go
package main

import (
	"fmt"
	"reflect"
)

type X struct{}
type Y struct{}

func main() {
	v := make(map[reflect.Kind]string)

	set := func(i interface{}, val string) {
		v[reflect.TypeOf(i).Kind()] = val
	}

	get := func(i interface{}) string {
		return v[reflect.TypeOf(i).Kind()]
	}

	set("test", "A")
	set(&X{}, "B")
	set(&Y{}, "C")

	fmt.Println(get(&Y{}))

}
```